docs/topics/authentication.rst

```diff
@@ -0,0 +1,126 @@
+                subscription_id: 3287abc8-f98a-c678-3bde-326766fd3617
+                tenant: ABCDEFAB-1234-ABCD-1234-ABCDEFABCDEF
+                client_id: ABCDEFAB-1234-ABCD-1234-ABCDEFABCDEF
+                secret: XXXXXXXXXXXXXXXXXXXXXXXX
+                cloud_environment: AZURE_PUBLIC_CLOUD
+
+When the actual state block is called, the pillar is referenced:
+
+.. code-block:: jinja
+
+        {% set profile = salt['pillar.get']('azurerm:mysubscription') %}
+        Ensure resource group exists:
+            azurerm_resource.resource_group_present:
+                - name: my_rg
+                - location: westus
+                - tags:
+                    how_awesome: very
+                    contact_name: Elmer Fudd Gantry
+                - connection_auth: {{ profile }}
+
+UsernamePasswordCredential
+==========================
+The ``UsernamePasswordCredential`` authenticates a user using work and school accounts' 
+usernames and passwords (Microsoft accounts are not supported). This credential was 
+updated from msrestazure's ``UserPassCredentials``, also maintaining functionality while 
+being compatible with updated dependencies. This is the second authentication option in 
+the code and requires a ``username``, ``password``, and ``. This type of authentication 
+is not as recommended as the other types because it is not as secure or compatible with 
+other functionalities (such as multi-factor authentication and consent prompting). This 
+information can be provided to Salt through the testing file, similar to the example 
+above, but with changed variables. This information is also passed through Salt by pillar 
+in the ``test.sls`` file:
+
+.. code-block:: yaml
+
+        azurerm:
+            user_pass_auth:
+                subscription_id: 3287abc8-f98a-c678-3bde-326766fd3617
+                username: fletch
+                password: 123pass
+
+Similarly, the pillar is referenced:
+
+.. code-block:: jinja
+
+        {% set profile = salt['pillar.get']('azurerm:user_path_auth') %}
+        Ensure resource group exists:
+            azurerm_resource.resource_group_present:
+                - name: my_rg
+                - location: westus
+                - tags:
+                    how_awesome: very
+                    contact_name: Elmer Fudd Gantry
+                - connection_auth: {{ profile }}
+
+
+
+DefaultAzureCredential
+======================
+The ``DefaultAzureCredential`` authenticates  when no other specifications are provided. 
+This credential was updated from msrestazure's ``MSIAuthentication``, so it now allows 
+the user to get the access tokens rather than just setting them. This type of 
+authentication is the last and default authentication option. Based on different 
+situations, DefaultAzureCredential automatically goes through multiple different 
+mechanisms and detects the best fit authentication method:
+
+#. Environment: Authenticates using environment variables.
```

Comment:
**suggestion (blocking):** Might be good to link out to the environment variables that people can use.

`https://docs.microsoft.com/en-us/python/api/azure-identity/azure.identity.environmentcredential?view=azure-python`